sphinxcontrib/needs/directives/need.py

```diff
@@ -138,11 +142,21 @@ def run(self):
             env.need_all_needs = {}
 
         if id in env.need_all_needs.keys():
-            raise NeedsDuplicatedId("A need with ID {0} already exists! This is not allowed".format(id))
+            if 'id' in self.options:
+                raise NeedsDuplicatedId("A need with ID {0} already exists! "
+                                        "This is not allowed".format(id))
+            else:  # this is a generated ID
+                raise NeedsDuplicatedId(
+                    "Needs could not generate a unique ID for a need with "
+                    "the title '{}' because another need had the same title. "
+                    "Either supply IDs for the requirements or ensure the "
+                    "titles are different.  NOTE: If title is being generated "
+                    "from the content, then ensure the first sentence of the "
+                    "requirements are different.".format(' '.join(self.content)))
```

Comment:
...format(' '.join(self.content))) -->
Shouldn't this be self.trimmed_title?
Content could be quite long...

Reply:
It’s wrong, but I think it should be self.full_title as that’s what gets hashed. If we just show trimmed, then there’s a small chanced it might be ambiguous which requirements caused the collision.

Reply:
It’s wrong, but I think it should be self.full_title as that’s what gets hashed. If we just show trimmed, then there’s a small chanced it might be ambiguous which requirements caused the collision.

Reply:
It’s wrong, but I think it should be self.full_title as that’s what gets hashed. If we just show trimmed, then there’s a small chanced it might be ambiguous which requirements caused the collision.

Reply:
Fixed and pushed an update.